01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+summary(share_households_place)
+```
+
+```{r}
+#| label: check-share-households-place-age
+
+stopifnot(min(share_households_place_subgroup$share_households) >= 0)
+stopifnot(max(share_households_place_subgroup$share_households) <= 1)
+
+share_households_place_subgroup |>
+  ggplot(aes(x = share_households)) +
+  geom_histogram(color = "blue", fill = alpha("blue", 0.3), binwidth = 0.02) +
+  facet_wrap(subgroup ~ race_ethnicity, ncol = 2) +
+  labs(
+    title = "Share of Households by Race and Ethnicity",
+    subtitle = "Place",
+    y = "Count",
+    x = "Share of Households"
+  )
+
+skimr::skim(share_households_place_subgroup)
+summary(share_households_place_subgroup)
+```
+
+:::
+
+### Housing Value Share
+
+We keep observations for homeowners only in order to calculate total aggregated housing wealth for homeowners for each race and ethnic group divided by the total aggregated housing value
+
+```{r}
+#| label: total-house-values-by-race
+
+house_value_by_race <- function(data, ...) {
+  data |>
+    mutate(ownership = as_factor(ownership)) |>
+    filter(ownership == "Owned or being bought (loan)") |>
+    summarise(
+      house_value = sum(valueh * hhwt),
+      home_owners = n(),
+      .by = c(year, state, race_ethnicity, ...)
+    )
+}
+
+house_value_county_subgroup <- house_value_by_race(households_county, county, subgroup)
+house_value_county <- house_value_by_race(households_county, county)
+
+
+house_value_place_subgroup <- house_value_by_race(households_place, place, subgroup)
+house_value_place <- house_value_by_race(households_place, place)
+```
+
+
+The metrics for the ACS indices are based on:
+- (1) sample size of each race and ethnic household at the county/place and 
+- (2) the number of observations that comes from the specific county/place calculated from the PUMA-county/place crosswalk.
+
+Therefore, the data quality is poor for counties/places with a extremely small number of households (N). To assess the data quality of counties/places with larger N, we need to consider what percent of data for the county/place actually came from the county/place itself.
+
+In particular,
+
+-   Data quality is good (`1`): N is greater than 30 and more than 75% of observations are from the county/place.
+-   Data quality is marginal (`2`): N is greater than 30 but only 35% to 75% of observations are from the county/place.
+-   Data quality is poor (`3`): N is less than 30 **or** if less than 35% of observations are from the county/place
+
+```{r}
+#| label: quality-check-flag-functions
+
+construct_quality_flag <- function(x, y) {
+  case_when(
+    x < 30 ~ 3,
+    x >= 30 & y < 0.35 ~ 3,
+    x >= 30 & y >= 0.35 & y < 0.75 ~ 2,
+    x >= 30 & y >= 0.75 ~ 1
+  )
+}
+```
+
+::: {.panel-tabset}
+
+### County
+
+```{r}
+#| label: add-quality-check-flags-county
+
+county_quality <-
+  households_county |>
+  distinct(year, state, county, .keep_all = TRUE) |>
+  select(year, state, county, sum_products)
```

Comment:
sum_products is from the crosswalk-- I might note where the crosswalk documentation can be found in the text introducing the code where you start using the crosswalk, so folks can look up what this variable is.

Reply:
Added a link to the specific notebook that describes this from the crosswalk folder!

Reply:
Thank you!